### PR TITLE
Seeing Food Makes You Hungry

### PR DESCRIPTION
**Pull request by ss13player**
[Changelogs]: # 

:cl: ss13 player
add: You get hungry from looking at food
/:cl:

[why]: # People don't get hungry a lot, this adds more tension and makes it more realistic.

### CODE REVIEW

**Comment by WJohn** [comment]
New torture device: Surround a cuffed man with delicious food that he *can't eat*

**Comment by ss13player** [comment]
i might change that if theres demand for it

**Comment by optimumtact** [comment]
what's your in game ckey

**Comment by ss13player** [comment]
Megahammer

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 1]
```diff
@@ -14,11 +14,15 @@
 	taste_mult = 4
 	var/nutriment_factor = 1 * REAGENTS_METABOLISM
 
+/mob/living/carbon/var/last_eaten = 0
```
Comment (optimumtact) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: lol

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -311,8 +311,20 @@
 	// Tissues die without blood circulation
 	adjustBruteLoss(2)
 
+/mob/living/carbon/human/var/last_seen_nearby_food = 0
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/life.dm]: also lol

**Comment by ss13player** [comment]
i fixed it

File: code/modules/mob/living/carbon/human/life.dm [hunk 3]
```diff
@@ -311,8 +311,18 @@
...
 
-
-
+/mob/living/carbon/human/proc/handle_nearby_food()
+	if(!client) return
+	if(world.time - last_eaten < 30 SECONDS) return //30 seconds after eating food you start getting hungry from seeing food again
+	if(world.time - last_seen_nearby_food < 15 SECONDS) return
+	if(nutrition >= NUTRITION_LEVEL_WELL_FED) return
+
+	for(var/obj/item/reagent_containers/food/F in view(client.view, get_turf(src)))
```
Comment (kevinz000) [inline code/modules/mob/living/carbon/human/life.dm]: i'm not qualified (see: maintainer) to be complaining about this stuff but is calling view on a 15x15 area just to find food every second for every connected human player really worth a meme feature?

**Comment by kevinz000** [comment]
also hi mrty

File: code/modules/mob/living/carbon/human/life.dm (hunk 3 above)
Comment (optimumtact) [reply code/modules/mob/living/carbon/human/life.dm]: no but we can leave this PR open for a while to indulge it

**Comment by ExcessiveUseOfCobblestone** [comment]
MRTY's signature is having cats embedded into his alts, he can't resist.

File: code/modules/mob/living/carbon/human/life.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/life.dm]: food is costly too, have you seen grocery prices these days (outside the cat food aisle)???

**Comment by ss13player** [comment]
i'm not the alt of some one named Morty

**Comment by Qustinnus** [comment]
How's that shadowban from goonchat going mrty

**Comment by ninjanomnom** [comment]
Don't torture the poor guy with something that's never getting merged. This is definitely not worth the cost.

**Pull request closed without merging**